odl/solvers/iterative/statistical.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Maximum Likelihood Expectation Maximization algorithm."""
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+import numpy as np
+
+__all__ = ('mlem', 'loglikelihood')
+
+
+AVAILABLE_MLEM_NOISE = ('poisson',)
+
+
+def mlem(op, x, rhs, niter=1, noise='poisson', callback=None):
+
+    """Maximum Likelihood Expectation Maximation algorithm.
+
+    Attempts to solve::
+
+        min_x L(x | rhs)
```

Comment:
Two glitches here: 1. The likelihood is maximized. 2. The arguments are switched. One maximizes the likelihood of observing the given data conditioned on the "parameters" `x`, which is `L(rhs | x)`.
Apart from that, the name `rhs` for "right-hand side" does not really make sense here since there is no equation. Perhaps we should switch to `data` across the library.